### DISCUSSION TITLE
Presence detection not working reliably

### DISCUSSION THREAD

**Discussion by petehhhhh** [tesla-local-control/tesla_ble_mqtt_docker#64] [Q&A]
Thanks so much for this project. Has been a godsend for my Tesla solar charging HA automation post Tesla restricting access to charge commands....

I am having one repeated issue. Approx every 2nd time i come home, presence won't recognise my car is "home", reporting it still as "away". i have tried getting using the update state button in mqqt  but generally does not help (did once).  If i restart the container it immediately picks it up. Have waited upto 2 hours for it to detect.

Any ideas how i can fix ?

A log from the most recent time i checked logs and then restarted the container follow. It is running on a Pi4 using docker-compose, talking back to HA running on a separate PI 5 over MQQT. All on the same network.

docker logs:

vin:LRWYHCEL3PC727266 presence has expired, set presence OFF
vin:LRWYHCEL3PC727266 presence has expired, set presence OFF
vin:LRWYHCEL3PC727266 presence has expired, set presence OFF
vin:LRWYHCEL3PC727266 presence has expired, set presence OFF
vin:LRWYHCEL3PC727266 presence has expired, set presence OFF
vin:LRWYHCEL3PC727266 presence has expired, set presence OFF
[09:21:15] Starting... loading /app/env.sh
[09:21:15] loading libproduct.sh
Setting up MQTT clients with authentication
Configuration Options are:
  BLE_CMD_RETRY_DELAY=5
  DEBUG=false
  MQTT_SERVER=ha.csure.me
  MQTT_PORT=1883
  MQTT_PASSWORD=Not Shown
  MQTT_USERNAME=mqtt
  PRESENCE_DETECTION_LOOP_DELAY=120
  PRESENCE_DETECTION_TTL=240
  TEMPERATURE_UNIT_FAHRENHEIT=false
  VIN_LIST=LRWYHCEL3PC727266
  MAX_CURRENT=48
  ENABLE_HA_FEATURES=true
Removing single buttons to be replaced by switches & covers:
windows, charger, cherge-port, climate, trunk
delete_legacies_singles; deleting legacy single MQTT entities topics
vin:LRWYHCEL3PC727266 presence has expired, set presence OFF
setupHADiscoveryAllVINsMain; Discarding any unread MQTT messages for topic:tesla_ble/LRWYHCEL3PC727266/config

**Comment by iainbullock** [comment]
Others have reported such behaviour. I'm away at the moment, but will try it out when I return and get back to you. Does it depend how long you are away from home before this occurs? Also what Bluetooth hardware are you using?

Thanks

**Reply by petehhhhh** [reply]
Many thanks. 

No doesn't seem to be time dependent but couldn't say for sure. 

It is running on the native Bluetooth hardware in a Pi 4. 

Let me know if I can help in any way...

**Comment by iainbullock** [comment]
I'm running Raspberry Pi 3 with built in bluetooth. I will test it when I get home next week to see if I get the same issue

**Comment by petehhhhh** [comment]
It’s a bit random unfortunately eg yesterday worked fine. On 24 Oct 2024, at 2:38 am, Iain Bullock ***@***.***> wrote:﻿
I'm running Raspberry Pi 3 with built in bluetooth. I will test it when I get home next week to see if I get the same issue

**Comment by iainbullock** [comment]
I've been monitoring mine for last 2 days. The car was away for 13 hours yesterday, presence worked as expected. I will keep monitoring
